Receptai.client/pages/user/profile.vue

```diff
@@ -1,11 +1,210 @@
+  name: string | null;
+  email: string | null;
+  avatarUrl: string | null;
+}
+
+interface UserCookie {
+  token: string;
+  expiresIn: number;
+  user: User;
+}
+
+const user: User = reactive({
+  id: null,
+  name: null,
+  email: null,
+  avatarUrl: null,
+});
+const showConfirmation = ref(false);
+const TastyBytes_user = useCookie<UserCookie | null>("TastyBytes_user");
+
+const validationSchema = toTypedSchema(
+  zod.object({
+    name: zod
+      .string()
+      .min(3, "Name must be at least three characters long.")
+      .regex(
+        /^[a-zA-Z]{3,}(?: [a-zA-Z]+){0,2}$/,
+        "Name can't contain special symbols."
+      ),
+  })
+);
+
+const { handleSubmit, errors } = useForm({
+  validationSchema,
+});
+
+const { value: name } = useField("name");
+
+if (TastyBytes_user.value) {
+  axios
+    .get(`${config.public.baseURL}/api/v1/user/me`, {
+      headers: { Authorization: `Bearer ${TastyBytes_user.value.token}` },
+    })
+    .then((d) => {
+      user.id = d.data.id;
+      name.value = d.data.name;
+      user.name = d.data.name;
+      user.email = d.data.email;
+      user.avatarUrl = d.data.avatarUrl;
+    });
+} else {
+  TastyBytes_user.value = null;
+}
+
+const onSubmit = handleSubmit(async () => {
+  showConfirmation.value = false;
+
+  if (name.value === user.name) {
+    return;
+  }
+
+  if (TastyBytes_user.value) {
+    try {
+      await axios.patch(
+        `${config.public.baseURL}/api/v1/user/edit/${TastyBytes_user.value.user.id}`,
+        {
+          newName: name.value,
+        },
+        {
+          headers: { Authorization: `Bearer ${TastyBytes_user.value.token}` },
+        }
+      );
+
+      const newObject = TastyBytes_user.value;
+      newObject.user.name = name.value as string;
+      user.name = newObject.user.name;
+      TastyBytes_user.value = newObject;
+      addNotification(
+        "Name change was successful! Keep in mind that your change might take some time to show everywhere.",
+        "Success"
+      );
+    } catch (e) {
+      console.log(e);
+
+      name.value = user.name;
+      addNotification(
+        "There was an error when changing your name. Please try again.",
+        "Error"
+      );
+    }
+  }
+});
+
+const onCancel = () => {
+  showConfirmation.value = false;
+  name.value = user.name;
+};
 </script>
 
 <template>
-  <div>Profile</div>
+  <div v-if="user.name !== null">
+    <div>
+      <h1 class="text-3xl font-bold text-center m-5">Profile</h1>
+    </div>
+    <div v-if="user">
+      <div class="flex flex-col justify-center items-center">
+        <div
+          class="w-full py-10 flex justify-center items-center rounded-md bg-cover bg-[url(https://th.bing.com/th/id/R.3d35003509104e7b7e0b8e2ac4d1ca2c?rik=dYArdd5RRivs9A&riu=http%3a%2f%2fwww.yurtopic.com%2ffood%2fcooking%2fimages%2fworld-of-gourmet%2fDelicious-field-salad2.jpg&ehk=2NTSPZUV8VCbauRbIZ6stV3ClmYd%2bTHctu7uswVH%2fIk%3d&risl=&pid=ImgRaw&r=0)] shadow-[inset_0_0_0_1000px_#0000004f] bg-center"
```

Comment:
This URL wherever it leads (probably an image) should be extracted as a standalone image and stored locally.

Reply:
The thinking behind this was to have an array of images that would change between each session. But I see your point, all the images can be stored locally as they will change infrequently.